droidlet/memory/memory_filters.py

```diff
@@ -463,22 +463,31 @@ def handle_output(self, agent_memory, query, memids):
                 except:
                     raise Exception("malformed output clause: {}".format(query))
             values_dict = {m: [] for m in memids}
-            for aname in attribute_name_list:
-                if type(aname) is not str:
-                    raise Exception(
-                        "output attribute in basic search should be (list of) simple properties, instead got: {}".format(
-                            attribute_name_list
+            for m in memids:
+                attributes = []
+                for aname in attribute_name_list:
+                    if type(aname) is not str:
+                        raise Exception(
+                            "output attribute in basic search should be (list of) simple properties, instead got: {}".format(
+                                attribute_name_list
+                            )
                         )
-                    )
-                for m in memids:
-                    values_dict[m].append(get_property_value(agent_memory, m, aname))
-            if len(attribute_name_list) == 1:
-                for m in values_dict:
-                    values_dict[m] = values_dict[m][0]
+                    prop_vals = get_property_value(agent_memory, m, aname, get_all)
+                    if type(prop_vals) is list:
+                        for prop_val in prop_vals:
+                            attributes.append(prop_val)
+                    else:
+                        attributes.append(prop_vals)
+                if len(attribute_name_list) == 1:
+                    # attributes for this memid are single values
+                    values_dict[m] += attributes
+                else:
+                    # attributes for this memid are a list
+                    values_dict[m].append(attributes)
```

Comment:
I think this is correct. If there are multiple attribute names, then the props are a list of lists (one for each attribute name). For example, (x,y) coords have two attributes in attribute_name_list: [x,y].